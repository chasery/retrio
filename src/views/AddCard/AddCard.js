import React, { useState } from 'react';
import { useHistory } from 'react-router-dom';
import Header from '../../components/Header/Header';
import Form from '../../components/Form/Form';
import FormField from '../../components/FormField/FormField';
import Error from '../../components/Error/Error';
import './AddCard.css';

function AddCard(props) {
  const history = useHistory();
  const [category, setCategory] = useState('');
  const [headline, setHeadline] = useState('');
  const [text, setText] = useState('');
  const [error, setError] = useState(null);

  const handleAddCard = (e) => {
    e.preventDefault();
  };

  return (
    <>
      <Header />
      <main role='main'>
        <section className='AddCard'>
          <div className='AddCard__wrapper'>
            <Form id='AddCard' onSubmit={(e) => handleAddCard(e)}>
              <div className='Form__header'>
                <h2>Add Card</h2>
              </div>
              <div className='Form__body'>
                <p>
                  Add a card to the{' '}
                  <strong>Team Name Pending Retrio Board</strong>.
                </p>
                <div className='FormField'>
                  <label className='FormField__label' htmlFor='category'>
                    Category<span className='FormField__required'>*</span>
                  </label>
                  <div class='FormField__select'>
                    <select
                      id='category'
                      required
                      value={category}
                      onChange={(e) => setCategory(e.target.value)}
                    >
                      <option value=''>Select Category</option>
                      <option value='1'>What wen't well</option>
                      <option value='2'>What didn't go well</option>
                      <option value='3'>To try</option>
                      <option value='4'>Shout outs</option>
                    </select>
                  </div>
                </div>
                <FormField
                  id='headline'
                  label='Headline'
                  type='text'
                  isRequired={true}
                  onChange={(e) => setHeadline(e.target.value)}
                  value={headline}
                />
                <div className='FormField'>
                  <label className='FormField__label' htmlFor='text'>
                    Text<span className='FormField__required'>*</span>
                  </label>
                  <textarea
                    id='text'
                    onChange={(e) => setText(e.target.value)}
                    rows='8'
                  ></textarea>
                </div>
                {error ? <Error message={error} /> : null}
                <div className='Form__controls'>
                  <button
                    className='Form__button secondary'
                    type='button'
                    onClick={() => history.push(`/boards/1`)}
                  >
                    Cancel
                  </button>
                  <button className='Form__button primary' type='submit'>
                    Add Card
                  </button>
                </div>
              </div>
            </Form>
          </div>
        </section>
      </main>
    </>
  );
}

export default AddCard;
