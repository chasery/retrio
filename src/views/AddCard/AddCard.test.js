import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import AddCard from './AddCard';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useParams: () => ({
    boardId: '800f3047-0054-4146-affsb7a-e06bf86f0868',
  }),
}));

describe('AddCard component', () => {
  it('renders without error', () => {
    const wrapper = shallow(<AddCard />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
