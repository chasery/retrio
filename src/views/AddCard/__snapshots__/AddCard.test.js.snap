// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCard component renders without error 1`] = `
<Fragment>
  <Header />
  <main
    role="main"
  >
    <section
      className="AddCard"
    >
      <div
        className="AddCard__wrapper"
      >
        <Form
          id="AddCard"
          onSubmit={[Function]}
        >
          <div
            className="Form__header"
          >
            <h2>
              Add Card
            </h2>
          </div>
          <div
            className="Form__body"
          >
            <p>
              Add a card to the
               
              <strong>
                Team Name Pending Retrio Board
              </strong>
               retrospective.
            </p>
            <div
              className="FormField"
            >
              <label
                className="FormField__label"
                htmlFor="category"
              >
                Category
                <span
                  className="FormField__required"
                >
                  *
                </span>
              </label>
              <div
                class="FormField__select"
              >
                <select
                  id="category"
                  onChange={[Function]}
                  required={true}
                  value=""
                >
                  <option
                    value=""
                  >
                    Select Category
                  </option>
                  <option
                    value="1"
                  >
                    What wen't well
                  </option>
                  <option
                    value="2"
                  >
                    What didn't go well
                  </option>
                  <option
                    value="3"
                  >
                    To try
                  </option>
                  <option
                    value="4"
                  >
                    Shout outs
                  </option>
                </select>
              </div>
            </div>
            <FormField
              id="headline"
              isRequired={false}
              label="Headline"
              onChange={[Function]}
              type="text"
              value=""
            />
            <div
              className="FormField"
            >
              <label
                className="FormField__label"
                htmlFor="text"
              >
                Text
                <span
                  className="FormField__required"
                >
                  *
                </span>
              </label>
              <textarea
                id="text"
                onChange={[Function]}
                rows="8"
              />
            </div>
            <div
              className="Form__controls"
            >
              <button
                className="Form__button secondary"
                onClick={[Function]}
                type="button"
              >
                Cancel
              </button>
              <button
                className="Form__button primary"
                type="submit"
              >
                Add Card
              </button>
            </div>
          </div>
        </Form>
      </div>
    </section>
  </main>
</Fragment>
`;
